src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx

```diff
@@ -251,82 +266,99 @@ export default class OrganizationDiscover extends React.Component {
-              multiple={true}
-              placeholder={this.getSummarizePlaceholder()}
-              options={fieldOptions}
-              value={currentQuery.fields}
-              onChange={val => this.updateField('fields', val.map(({value}) => value))}
-              clearable={true}
-            />
-          </Fieldset>
-          <Fieldset>
-            <Aggregations
-              value={currentQuery.aggregations}
-              columns={columns}
-              onChange={val => this.updateField('aggregations', val)}
-            />
-          </Fieldset>
-          <Fieldset>
-            <Conditions
-              value={currentQuery.conditions}
-              columns={columnsForConditions}
-              onChange={val => this.updateField('conditions', val)}
-            />
-          </Fieldset>
-          <Fieldset>
-            <SidebarLabel htmlFor="orderby" className="control-label">
-              {t('Order by')}
-            </SidebarLabel>
-            <SelectControl
-              name="orderby"
-              label={t('Order By')}
-              placeholder={<PlaceholderText>{t('Order by...')}</PlaceholderText>}
-              options={this.getOrderbyOptions()}
-              value={currentQuery.orderby}
-              onChange={val => this.updateField('orderby', val.value)}
-            />
-          </Fieldset>
-          <Fieldset>
-            <NumberField
-              name="limit"
-              label={<SidebarLabel>{t('Limit')}</SidebarLabel>}
-              placeholder="#"
-              value={currentQuery.limit}
-              onChange={val =>
-                this.updateField('limit', typeof val === 'number' ? val : null)}
+          {!showSavedQueries && (
+            <React.Fragment>
+              <SidebarHeader>
+                <SidebarTitle>{t('Query')}</SidebarTitle>
+                <Flex>
+                  <Box mr={1}>
+                    <Button size="xsmall" onClick={this.reset}>
+                      {t('Reset')}
+                    </Button>
+                  </Box>
+                  <Button
+                    size="xsmall"
+                    onClick={this.runQuery}
+                    priority="primary"
+                    busy={isFetchingQuery}
+                  >
+                    {t('Run')}
+                    {isFetchingQuery && <ButtonSpinner />}
+                  </Button>
+                </Flex>
+              </SidebarHeader>
+              <Fieldset>
+                <SidebarLabel htmlFor="fields" className="control-label">
+                  {t('Summarize')}
+                </SidebarLabel>
+                <SelectControl
+                  name="fields"
+                  multiple={true}
+                  placeholder={this.getSummarizePlaceholder()}
+                  options={fieldOptions}
+                  value={currentQuery.fields}
+                  onChange={val =>
+                    this.updateField('fields', val.map(({value}) => value))}
```

Comment:
Is `val` always an array even if the field is empty?

Reply:
Yes, the multi select dropdown type is always an array